### ISSUE TITLE
Lost MIDI2LR to LR communication and Crash issues

### ISSUE THREAD

**Issue by DavidWeaverPhoto** [rsjaffe/MIDI2LR#35]
It's been a frustrating 4 days.  After upgrading to 0.4.2 (might have nothing to do with it) I lost all apparent communication between LR (2015.1.1) and MIDI2LR.    After futzing around with it the last few days I finally reset the BCF2000 to original factory settings.  I removed the plug in using LR Plug-in Manager and rebooted, deleted the plug in from the LR plugin location,  downloaded a second copy of 0.4.2, then added it back.  Restated LR and nothing (nothing=no communication between LR and BCF.
Using BC Manager I verified that I'm using MIDI devices input and output are BCF2000 port 1, then I perform a hardware test which works.  Loading BCF2000.txt works.
MIDI2LR reports it is connected to LR but it does nothing in pick up mode.  There seems to be nothing going on between MIDI2LR/LR and the BCF2000.
AND MIDI2LR crashes when I "Rescan MIDI devices".

**Comment by rsjaffe** [comment]
I'm unsure of what the problem is. I'm working with LR 2015.2.1 and it's working just fine. I suggest rechecking your LR version. I had a terrible time with the 2015.2 update crashing everything, but it settled down a few days after going to .2.1 (I'm not sure what was going on initially, but it was clear that LR was having problems--it kept on doing the weekly backup every time I launched it). I also encountered issues with .2 that were only fixed by rebooting the computer.

Also, since I recently took over development of MIDI2LR, I have intentionally avoided working on anything other than menu choices. I have not yet changed anything that would change the basic workings of the program (e.g., MIDI message handling, user interface), as I've been waiting until I am more familiar with the software frameworks involved and more comfortable with the debugging challenges.

So, my suggestion is to:
1. Recheck version.
2. Go back to version 0.4.1 of MIDI2LR (download from https://github.com/rsjaffe/MIDI2LR/releases)
3. Reboot machine
3. If problems persist, it's LR.
4. If 0.4.1 works, then go to version 0.4.2. Then reboot machine. If problems show up at that point, we'll need to take a closer look at what was changed in 0.4.2, as well as look at at your specific configuration.

**Comment by DavidWeaverPhoto** [comment]
It's working now.  I think the cause was an upgrade to BC Manager which does not carry over the settings from a prior version.  I ran BC Manager in as an administrator to make sure settings were actually being written back.  I had to go back and set input/output not only under MIDI devices, I ALSO ahd to set MIDI input port and MIDI output port via BC Manager View--> B-Controls then from the B-Controls window MIDI-->Options to declare the ports for THAT specific controller.   This makes sense if  you work with MIDI devices all the time (Something few photographers probably do).   Then I closed BC Manager, rebooted and all seems to be working with MIDI2LR 0.4.2 and LR 2015.1.1.

Many thanks for your work on this.
